src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/ChainPal.Windows.cs

```diff
@@ -18,8 +18,10 @@ private ChainPal(SafeX509ChainHandle chain)
 
         internal static partial IChainPal FromHandle(IntPtr chainContext)
         {
-            if (chainContext == IntPtr.Zero)
-                throw new ArgumentNullException(nameof(chainContext));
+            unsafe
+            {
+                ArgumentNullException.ThrowIfNull((void*)chainContext, nameof(chainContext));
```

Comment:
Can we expose the `ArgumentNullException.ThrowIfNull(IntPtr)` overload publicly rather than require a pointer cast here?
Same comment in StorePal.\*.cs.

Reply:
(Hmm, I thought I'd actually undone my changes there... I guess they didn't stick.)

We _could_.  Especially with IntPtr and nint being one in the same, I'm not sure we want to new APIs that special-case IntPtr.Zero.  I'd expect moving forward any new APIs we provide that are pointer-based will actually take pointers, at which point we're mostly talking about doing so only for existing surface area. So, I'm not against another overload there, I'm just not sure it's worthwhile.

In the meantime, I'll revert the changes I'd made there as I'd intended to.